### ISSUE TITLE
backward compatibility in fn ptr structure

### ISSUE THREAD

**Issue by tushar-ty** [opencomputeproject/SAI#1109]
While adding incremental attributes, enums towards the end of the existing list, backward compatibility could further be easily achieved by adding new fns pts to the end of the structure list.

i.e

    sai_create_port_serdes_fn              create_port_serdes;
    sai_remove_port_serdes_fn              remove_port_serdes;
    sai_set_port_serdes_attribute_fn       set_port_serdes_attribute;
    sai_get_port_serdes_attribute_fn       get_port_serdes_attribute;
    sai_create_port_connector_fn           create_port_connector;
    sai_remove_port_connector_fn           remove_port_connector;
    sai_set_port_connector_attribute_fn    set_port_connector_attribute;
    sai_get_port_connector_attribute_fn    get_port_connector_attribute;

vs

    sai_create_port_connector_fn           create_port_connector;
    sai_remove_port_connector_fn           remove_port_connector;
    sai_set_port_connector_attribute_fn    set_port_connector_attribute;
    sai_get_port_connector_attribute_fn    get_port_connector_attribute;
    sai_create_port_serdes_fn              create_port_serdes;
    sai_remove_port_serdes_fn              remove_port_serdes;
    sai_set_port_serdes_attribute_fn       set_port_serdes_attribute;
    sai_get_port_serdes_attribute_fn       get_port_serdes_attribute;

**Comment by kcudnik** [comment]
yes, new functions should always be added to the end of existing structure
